lib/engines/content_block_manager/app/components/content_block_manager/content_block/document/show/embedded_objects/blocks_component.rb

```diff
@@ -61,13 +61,23 @@ def nested_blocks
   end
 
   def rows_for_nested_items(items, nested_name, index)
-    items.map do |key, value|
+    rows = items.map do |key, value|
       {
         key: key_to_title(key),
         value: content_for_row(embed_code_identifier(nested_name, index, key), value),
         data: data_attributes_for_row(embed_code_identifier(nested_name, index, key)),
       }
     end
+    ordered_by_field_order(rows, nested_name)
```

Comment:
Would it be easier to do the sorting before we map the items to rows? I'm just a bit concerned that the call to `parameterize` will mean some things get missed (especially if they're translated)